bundles/org.openhab.core.semantics/src/main/java/org/openhab/core/semantics/internal/SemanticsMetadataProvider.java

```diff
@@ -280,4 +289,28 @@ public void removed(Item item) {
     public void updated(Item oldItem, Item item) {
         processItem(item);
     }
+
+    class SemanticTagRegistryChangeListener implements RegistryChangeListener<SemanticTag> {
```

Comment:
That's not exactly what I had in mind, but it'll do here. Why is the class package-private and not private?

Reply:
You would like that I put it in a separate file ?